test/plugin_functional/test_suites/core_plugins/ui_plugins.ts

```diff
@@ -97,5 +98,45 @@ export default function({ getService, getPageObjects }: PluginFunctionalProvider
         ).to.be('/core_plugin_b/system_request says: "System request? false"');
       });
     });
+
+    describe('Static assets', function() {
+      describe('Platform assets', function() {
+        it('exposes static assets', async () => {
+          await supertest.get('/ui/favicons/favicon.ico').expect(200);
+        });
+
+        it('returns 404 if not found', async function() {
+          await supertest.get('/ui/favicons/not-a-favicon.ico').expect(404);
+        });
+
+        it('does not expose folder content', async function() {
+          await supertest.get('/ui/favicons/').expect(403);
+        });
+
+        it('does not allow file tree traversing', async function() {
+          await supertest.get('/ui/../../../../../README.md').expect(404);
+        });
+      });
+
+      describe('Plugin assets', function() {
+        it('exposes static assets from "public/assets" folder', async () => {
+          await supertest.get('/plugins/corePluginStaticAssets/assets/chart.svg').expect(200);
+        });
```

Comment:
Do we want to perform any test regarding the cache-related headers on these calls?

Reply:
good call, I was going to add, but forgot